### PR TITLE
Replace Apache Math library with own combinations generator

### PR DESCRIPTION
**Pull request by kenneth-fung**
- Remove Apache Commons math library
- Add `generateCombinations(List<T> items)` and `generateCombinations(List<T> items, int r)` to `CollectionUtil`
- Change checks for token input count to use `ArgumentMultimap#getValueCount`

### CODE REVIEW

File: src/main/java/budgetbuddy/commons/util/CollectionUtil.java [hunk 1]
```diff
@@ -50,4 +53,44 @@ public static boolean isAnyNonNull(Object... items) {
...
+        for (int i = 0; i < items.size(); i++) {
+            results.addAll(generateCombinations(items, i));
+        }
+        return results;
+    }
+
+    /**
+     * Returns a list of r-element combinations possible for a given list of items.
+     * @param r The number of elements for each combination.
+     * @return A {@code List} of size {@code r} {@code List}s of the given items.
```
Comment (tcgj) [inline src/main/java/budgetbuddy/commons/util/CollectionUtil.java]: Not sure what you mean by "a list of size r lists of the given items"

Comment (thwnus) [reply src/main/java/budgetbuddy/commons/util/CollectionUtil.java]: Probably "a list of size-`r` combinations of items from `items`".

**Comment by thwnus** [comment]
In the interest of time, merging first. Can tweak the documentation later on.

`technical_debt++`

File: src/main/java/budgetbuddy/commons/util/CollectionUtil.java (hunk 1 above)
Comment (kenneth-fung) [reply src/main/java/budgetbuddy/commons/util/CollectionUtil.java]: Yea, that's correct. I'll tweak it in the future.

**Pull request merged**